Lib/pydoc.py

```diff
@@ -389,6 +389,9 @@ def synopsis(filename, cache={}):
 class ErrorDuringImport(Exception):
     """Errors that occurred while trying to import something to document it."""
     def __init__(self, filename, exc_info):
+        if not isinstance(exc_info, tuple):
+            assert isinstance(exc_info, BaseException)
+            exc_info = type(exc_info), exc_info, exc_info.__traceback__
         self.filename = filename
         self.exc, self.value, self.tb = exc_info
```

Comment:
Separate suggestions:
1. Keep handling of two args in order.  First first and second second.  Handling first in middle of second is unnecessary confusion.
2. Handle two cases of second separately.  Don't create tuple just to unpack.  Unpacking makes it slightly more obvious that same attributes are defined, but pack/unpack makes it harder to see what object is assigned to each attribute.
```suggestion
        self.filename = filename
        if isinstance(exc_info, tuple):
            self.exc, self.value, self.tb = exc_info
        else:
            assert isinstance(exc_info, BaseException)
            self.exc = type(exc_info)
            self.value = exc_info
            self.tb = exc_info.__traceback__
```
I am tempted to think that no user will ever raise this exception. pydoc is only documented as a command-line app, not as importable library.   But this is not idlelib.  Can/should we emit a DeprecationWarning in the isinstance clause? It could then be removed someday.  Since there is no doc of the module content, no changed notice is needed.